includes/discount-actions.php

```diff
@@ -16,74 +18,121 @@
+	// Setup the discount code details
+	$posted = array();
 
-		foreach ( $data as $key => $value ) {
-			if ( $key != 'edd-dcg-discount-nonce' && $key != 'edd-action' && $key != 'edd-redirect' ) {
-				if ( is_string( $value ) || is_int( $value ) )
-					$posted[ $key ] = strip_tags( addslashes( $value ) );
-				elseif ( is_array( $value ) )
-					$posted[ $key ] = array_map( 'absint', $value );
-			}
+	foreach ( $data as $key => $value ) {
+		if ( in_array( $key, array( 'edd_dcg_discount_nonce', 'edd-action', 'edd-redirect' ), true ) ) {
+			continue;
 		}
+		if ( is_string( $value ) || is_int( $value ) ) {
+			$posted[ $key ] = strip_tags( addslashes( $value ) );
+		} elseif ( is_array( $value ) ) {
+			$posted[ $key ] = array_map( 'absint', $value );
+		}
+	}
 
-		if (!isset($posted['number-codes'])) return;
+	if ( ! isset( $posted['number-codes'] ) ) {
+		return;
+	}
 
-		// Check number of codes is number and greater than 0
-		if (floor($posted['number-codes']) == $posted['number-codes'] && $posted['number-codes'] > 0) {
-			$code = $posted;
-			unset($code['number-codes']);
-			unset($code['code-type']);
-			unset($code['code-limit']);
+	// Check number of codes is number and greater than 0
+	if ( floor( $posted['number-codes'] ) == $posted['number-codes'] && $posted['number-codes'] > 0 ) {
+		$code = $posted;
+		unset( $code['number-codes'] );
+		unset( $code['code-type'] );
+		unset( $code['code-limit'] );
 
-			$result = true;
-			// Loop through and generate code, check code doesnt exist _edd_discount_code
-			for ($i = 1; $i <= $posted['number-codes']; $i++) {
-			    $code['name'] = $posted['name'] .'-'. $i;
-			    $code['code'] = edd_dcg_create_code($posted['code-type'], $posted['code-limit']);
-			    $code['status'] = 'active';
-			    $result = edd_store_discount( $code );
-			    if (!$result) break;
+		$fields_to_convert = array(
+			'products'   => 'product_reqs',
+			'min_price'  => 'min_charge_amount',
+			'max'        => 'max_uses',
+			'use_once'   => 'once_per_customer',
+			'start'      => 'start_date',
+			'expiration' => 'end_date',
+		);
+		if ( function_exists( 'edd_add_adjustment' ) ) {
+			$code['scope'] = ! empty( $data['not_global'] ) ? 'not_global' : 'global';
+			foreach ( $fields_to_convert as $edd2x => $edd30 ) {
+				$code[ $edd30 ] = ! empty( $code[ $edd2x ] ) ? $code[ $edd2x ] : '';
+				unset( $code[ $edd2x ] );
 			}
+		}
 
-			if ( $result ) {
-				$args = array( 	'edd-message' => 'discounts_added',
-								'edd-number' => $posted['number-codes'] );
-				wp_redirect( add_query_arg( $args , $data['edd-redirect'] ) ); edd_die();
+		$result = true;
+		// Loop through and generate code, check code doesnt exist _edd_discount_code
+		for ( $i = 1; $i <= $posted['number-codes']; $i++ ) {
+			$code['name']   = $posted['name'] . '-' . $i;
+			$code['code']   = edd_dcg_create_code( $posted['code-type'], $posted['code-limit'] );
+			$code['status'] = 'active';
+			if ( function_exists( 'edd_add_adjustment' ) ) {
```

Comment:
```suggestion
			if ( function_exists( 'edd_add_discount' ) ) {
```

Seems like this makes more sense.

Reply:
Unfortunately it does not, because `edd_add_discount` has been in EDD since the beginning (or 1.0), but was refactored for 3.0.

Reply:
3.0 lied to me because we have `@since 3.0` on that. Looks like we repurposed an existing function to do something else... annoying...